### ISSUE TITLE
Possible multi-threading bug?

### ISSUE THREAD

**Issue by KallyopeBio** [gephi/gephi#1950]
## Expected Behavior
Consistent performance and behavior every time I start Gephi and perform exactly the same actions.

## Current Behavior
I am loading a large network into Gephi (386K nodes and 9.3M edges). I load the data from the Data Laboratory tab, then I switch to the Overview tab. Now there are two separate things that happen inconsistently, when I load exactly the same network, perform exactly the same steps, and have no other programs running on my machine:
1.) Occasionally, Gephi claims that there are parallel edges in the edges spreadsheet. But most of the time, it does not. And I've checked: there aren't any parallel edges.
2.) In the best case scenario, the network (with its random initial node positions) will render in about 60 seconds after I switch to the Overview tab. In the worst case scenario, it never renders. I noticed the presence of a lock file in ~/.gephi/0.9.2/dev when the Overview loading appears to get stuck. If I delete that lock file, Gephi will render about a minute later. 

## Possible Solution

Is Gephi getting dead-locked somewhere during the loading of the Overview tab? The symptoms I described above are characteristic of a threading problem: a bug that doesn't happen every time; that happens during extensive parallelization (as during the import of a large edges spreadsheet); and that seems to be resolved by deleting a lock file. 

## Steps to Reproduce

1. Start Gephi 0.9.2
2. Click on the Data Laboratory tab.
3. First import a nodes spreadsheet. 
4. Then import an edges spreadsheet. Here's where the first bug appears: sometimes Gephi finds parallel edges in the spreadsheet, but most of the time it doesn't--even though I import exactly the same spreadsheet each time. 
5.) Once the edges are done being imported, click on the Overview tab. 

## Your Environment

* Version used: Gephi 0.9.2
* Java version: 1.8.0_161 (This is Oracle's version, not the openjdk)
* Operating System: Ubuntu 16.04

**Comment by KallyopeBio** [comment]
[messages.log](https://github.com/gephi/gephi/files/1876220/messages.log)

**Comment by KallyopeBio** [comment]
I've attached a log file from a slow run. The log file does not appear to differ between runs that do and do not detect parallel edges

**Comment by eduramiba** [comment]
Thanks for the report, could you share a file to reproduce the parallel edges problem? What file format do you use?

**Comment by KallyopeBio** [comment]
Absolutely. Both the nodes and edges files are plain text csv files. I've put both into a tar.gz, but it's still too large to upload here. How can I share it?

**Comment by eduramiba** [comment]
Thanks, you can use wetransfer or similar and send me a link to my email (my username at gmail).

**Comment by KallyopeBio** [comment]
OK, I just sent the files to your email address

**Comment by eduramiba** [comment]
Thanks, will take a look when I have some time.

**Comment by KallyopeBio** [comment]
OK, thanks. The data within are proprietary, so please don't share them or make it publicly available.

**Comment by KallyopeBio** [comment]
Actually, I removed the link to the proprietary data; I'll send you an anonymized version of the same data

**Comment by eduramiba** [comment]
Ok no problem!

**Comment by KallyopeBio** [comment]
OK, the anonymized data have been sent to your email address.
